sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -898,11 +899,15 @@ class DatasetSuite extends QueryTest with SharedSQLContext {
       (1, 2), (1, 1), (2, 1), (2, 2))
   }
 
-  test("dropDuplicates should not change child plan output") {
-    val ds = Seq(("a", 1), ("a", 2), ("b", 1), ("a", 1)).toDS()
-    checkDataset(
-      ds.dropDuplicates("_1").select(ds("_1").as[String], ds("_2").as[Int]),
-      ("a", 1), ("b", 1))
+  test("SPARK-19065 dropDuplicates should not create expressions using the same id") {
```

Comment:
how about we remove this test and add a new test to show the behavior change more obvious?
```
val df = ...
val df2 = df.dropDuplicates("i")
intercept[AnalysisException] { df2.select(df("i")) }
```